webapp-django/crashstats/topcrashers/views.py

```diff
@@ -0,0 +1,302 @@
+    params['_aggs.signature'] = [
+        'platform',
+        'is_garbage_collecting',
+        'hang_type',
+        'process_type',
+    ]
+
+    # We don't care about no results, only facets.
+    params['_results_number'] = 0
+
+    if params.get('process_type') in ('any', 'all'):
+        params['process_type'] = None
+
+    api = SuperSearchUnredacted()
+    search_results = api.get(**params)
+
+    if search_results['total'] > 0:
+        results = search_results['facets']['signature']
+
+        platforms = models.Platforms().get_all()['hits']
+        platform_codes = [
+            x['code'] for x in platforms if x['code'] != 'unknown'
+        ]
+
+        for i, hit in enumerate(results):
+            hit['signature'] = hit['term']
+            hit['rank'] = i + 1
+            hit['percent'] = 1.0 * hit['count'] / search_results['total']
+
+            # Number of crash per platform.
+            for platform in platform_codes:
+                hit[platform + '_count'] = 0
+
+            sig_platforms = hit['facets']['platform']
+            for platform in sig_platforms:
+                code = platform['term'][:3].lower()
+                if code in platform_codes:
+                    hit[code + '_count'] = platform['count']
+
+            # Number of crashes happening during garbage collection.
+            hit['is_gc_count'] = 0
+
+            sig_gc = hit['facets']['is_garbage_collecting']
+            for row in sig_gc:
+                if row['term'].lower() == 't':
+                    hit['is_gc_count'] = row['count']
+
+            # Number of plugin crashes.
+            hit['plugin_count'] = 0
+
+            sig_process = hit['facets']['process_type']
+            for row in sig_process:
+                if row['term'].lower() == 'plugin':
+                    hit['plugin_count'] = row['count']
+
+            # Number of hang crashes.
+            hit['hang_count'] = 0
+
+            sig_hang = hit['facets']['hang_type']
+            for row in sig_hang:
+                if row['term'] in (1, -1):
+                    hit['hang_count'] = row['count']
+
+        # Run the same query but for the previous date range, so we can
+        # compare the rankings and show rank changes.
+        dates = get_date_boundaries(params)
+        delta = (dates[1] - dates[0]) * 2
+        params['date'] = [
+            '>=' + (dates[1] - delta).isoformat(),
+            '<=' + dates[0].isoformat()
+        ]
+
+        previous_range_results = api.get(**params)
+        total = previous_range_results['total']
+
+        compare_signatures = {}
+        if (total > 0 and 'signature' in previous_range_results['facets']):
+            signatures = previous_range_results['facets']['signature']
+            for i, hit in enumerate(signatures):
+                compare_signatures[hit['term']] = {
+                    'count': hit['count'],
+                    'rank': i + 1,
+                    'percent': 1.0 * hit['count'] / total
```

Comment:
Do '100.0 * hit['count'] / total` here instead and it starts to make sense calling it "percent".